crema/confidence.py

```diff
@@ -199,11 +208,13 @@ def _compete(self, df, group_columns):
             keep = "first"
 
         group_columns = listify(group_columns)
-        out_df = (
-            df.sample(frac=1)  # This is so ties are broken randomly.
-            .sort_values(group_columns + [self._score_column])
-            .drop_duplicates(group_columns, keep=keep)
+        out_df = df.sample(
+            frac=1
+        ).sort_values(  # This is so ties are broken randomly.
+            [self._score_column] + group_columns
         )
+        for columns in group_columns:
```

Comment:
The reason it is there is because I need to call drop_duplicates on the different columns one by one. First I need to drop by peptide sequence. Then I need to drop by the peptide pairing. If I do it both at once, it will only drop rows where both sequence and pairing are the same - which ends up not dropping entries that should be dropped.